src/docs/arch-session.dox

```diff
@@ -0,0 +1,44 @@
+/*! @arch_page arch-session Session
+
+@section arch_session_def Definition
+After a @ref arch-connection has been established between the application and WiredTiger, the
+application can start sending requests to WiredTiger using a session. A session is internally
+represented by WT_SESSION and plays an important role since almost all operations require one.
+
+A session can only be created through an existing connection with the API
+WT_CONNECTION::open_session. It is possible to create multiple sessions through the same connection.
+In fact, one connection can have multiple sessions but one session can only be associated to one
+connection.
+
+@section arch_session_ops Operations
+The different operations a session can request to WiredTiger are related to cursors handles, table
+operations and transactions. You can read the complete description of each possible operation in the
+documentation related to WT_SESSION.
+
+@section arch_session_txn Transactions
+It is possible to group a number of operations within a session and get a single result, in other
```

Comment:
"get a single result" is a bit strange. I would simply say group them in a transaction.

Reply:
See 339e6d00b48db9f60bd28f04c4046825929dcb6a